File: frontend/src/types/nativeText.ts
import { OCRBBox } from "./ocr";

export interface NativeTextStyle {
  font_name: string;
  font_size: number;
  color: string | null;
  bold: boolean;
  italic: boolean;
}

export interface NativeTextBlock {
  id: string;
  text: string;
  bbox: OCRBBox;
  style: NativeTextStyle;
}

export interface NativeTextPageResult {
  page_number: number;
  page_width: number;
  page_height: number;
  coordinate_space: string;
  blocks: NativeTextBlock[];
}

File: backend/app/models/native_text.py
from typing import Optional

from pydantic import BaseModel

from app.models.ocr import OCRBBox


class NativeTextStyle(BaseModel):
    font_name: str
    font_size: float
    color: Optional[str] = None
    bold: bool = False
    italic: bool = False


class NativeTextBlock(BaseModel):
    id: str
    text: str
    bbox: OCRBBox
    style: NativeTextStyle


class NativeTextPageResult(BaseModel):
    page_number: int
    page_width: float
    page_height: float
    coordinate_space: str = "pdf_points"
    blocks: list[NativeTextBlock]
